### PR TITLE
TST: test (bleeding-edge) with Cython 3.0

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary

As explained in #4373, I expect this will fail to build on GHA.
I'm opening this as a draft so it can be used as a reference if I need to report the issue upstream

### CODE REVIEW

**Comment by neutrinoceros** [comment]
The remaining compilation error was identified as a bug in Cython 3.0.0b1 (adressed in https://github.com/cython/cython/pull/5317). It is also possible to work around it by requiring `-std=c++11` instead of `-std=c++03` for `cykdtree`. I think this is a reasonable thing to do given that C++11 was originally listed as a requirement in [the standalone distribution of the package](https://pypi.org/project/cykdtree/). Here's the entire patch
```patch
diff --git a/setup.py b/setup.py
index 8dd8238ba..8ec14a839 100644
--- a/setup.py
+++ b/setup.py
@@ -25,7 +25,7 @@ CPP14_CONFIG = defaultdict(
     lambda: check_CPP14_flags(["-std=c++14", "-std=c++1y", "-std=gnu++0x"]),
     {"msvc": ["/std:c++14"]},
 )
-CPP03_CONFIG = defaultdict(lambda: ["-std=c++03"], {"msvc": ["/std:c++03"]})
+CPP11_CONFIG = defaultdict(lambda: ["-std=c++11"], {"msvc": ["/std:c++11"]})

 _COMPILER = get_default_compiler()

@@ -37,7 +37,7 @@ else:
     std_libs = ["m"]

 CPP14_FLAG = CPP14_CONFIG[_COMPILER]
-CPP03_FLAG = CPP03_CONFIG[_COMPILER]
+CPP11_FLAG = CPP11_CONFIG[_COMPILER]

 cythonize_aliases = {
     "LIB_DIR": "yt/utilities/lib/",
@@ -53,7 +53,7 @@ cythonize_aliases = {
     "FIXED_INTERP": "yt/utilities/lib/fixed_interpolator.cpp",
     "ARTIO_SOURCE": glob.glob("yt/frontends/artio/artio_headers/*.c"),
     "CPP14_FLAG": CPP14_FLAG,
-    "CPP03_FLAG": CPP03_FLAG,
+    "CPP11_FLAG": CPP11_FLAG,
 }

 lib_exts = [
diff --git a/yt/utilities/lib/cykdtree/kdtree.pyx b/yt/utilities/lib/cykdtree/kdtree.pyx
index 5b7ecd598..9eac3118a 100644
--- a/yt/utilities/lib/cykdtree/kdtree.pyx
+++ b/yt/utilities/lib/cykdtree/kdtree.pyx
@@ -3,7 +3,7 @@
 # distutils: sources = yt/utilities/lib/cykdtree/c_utils.cpp
 # distutils: depends = yt/utilities/lib/cykdtree/c_kdtree.hpp, yt/utilities/lib/cykdtree/c_utils.hpp
 # distutils: language = c++
-# distutils: extra_compile_args = CPP03_FLAG
+# distutils: extra_compile_args = CPP11_FLAG
 import numpy as np

 cimport numpy as np
diff --git a/yt/utilities/lib/cykdtree/utils.pyx b/yt/utilities/lib/cykdtree/utils.pyx
index 276212c41..67c0edf9f 100644
--- a/yt/utilities/lib/cykdtree/utils.pyx
+++ b/yt/utilities/lib/cykdtree/utils.pyx
@@ -2,7 +2,7 @@
 # distutils: sources = yt/utilities/lib/cykdtree/c_utils.cpp
 # distutils: depends = yt/utilities/lib/cykdtree/c_utils.hpp
 # distutils: language = c++
-# distutils: extra_compile_args = CPP03_FLAG
+# distutils: extra_compile_args = CPP11_FLAG
 import numpy as np

 cimport numpy as np
 ```
 
 Any objections ? @matthewturk @Xarthisius @cphyc ?

**Comment by neutrinoceros** [comment]
I'll just push the patch to prove that it fixes the problem and open the PR for review

File: setup.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
     lambda: check_CPP14_flags(["-std=c++14", "-std=c++1y", "-std=gnu++0x"]),
     {"msvc": ["/std:c++14"]},
 )
-CPP03_CONFIG = defaultdict(lambda: ["-std=c++03"], {"msvc": ["/std:c++03"]})
+CPP11_CONFIG = defaultdict(lambda: ["-std=c++11"], {"msvc": ["/std:c++11"]})
```
Comment (matthewturk) [inline setup.py:28]: I agree, this seems the right thing to do.  Now that EWAH has been split out, do we still use C++14 anywhere?

**Review by matthewturk — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (neutrinoceros) [reply setup.py:28]: Ewah has been split out ?

Comment (matthewturk) [reply setup.py:28]: ignore me, I am all gutters no strikes today.

**Pull request merged**